File: ApplicationServices/Models/Flights/search/SegmentDto.cs
namespace ApplicationServices.Models.Flights.search
{
    public class SegmentDto
    {
        public string id { get; set; }
        public TravelDto departure { get; set; }
        public TravelDto arrival { get; set; }
        public string carrierCode { get; set; }
        public string number { get; set; }
        public string duration { get; set; }

        public int numberOfStops { get; set; }
        public bool blacklistedInE { get; set; }
    }
}

File: WebService/src/app/model/flights/segment.dto.ts
import { TravelDto } from './travel.dto';

export class SegmentDto {
  constructor(
    public departure: TravelDto,
    public arrival: TravelDto,
    public carrierCode: string,
    public number: string,
    public duration: number,
    public durationMinutes: number,
    public id: string,
    public numberOfStops: number,
    public blacklistedInEU: boolean
  ) {}
}
